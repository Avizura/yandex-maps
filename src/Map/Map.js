import React from 'react';
import PropTypes from 'prop-types';
import { YMaps, Map, Placemark, SearchControl, ZoomControl, FullscreenControl, Polyline } from 'react-yandex-maps';
import get from 'lodash/get';

const mapState = {
  center: [55.76, 37.64],
  zoom: 10,
  controls: []
};

const CustomMap = ({ placemarks, handleAddPlacemark, handleSelectSearchResult, addPlacemarkRef, handlePlacemarkDragEnd }) => (
  <YMaps>
    <Map width="100%" height={600} state={mapState} onClick={handleAddPlacemark}>
      <ZoomControl />
      <FullscreenControl />
      <SearchControl
        options={{
          size: 'auto',
          noPlacemark: true,
          noSelect: false,
          float: 'right'
        }}
        onResultSelect={handleSelectSearchResult}
      />
      {placemarks.map((placemarkParams, index) => (
        <Placemark
          key={index}
          instanceRef={addPlacemarkRef(index)}
          {...placemarkParams}
          options={{
            draggable: true
          }}
          onDragEnd={handlePlacemarkDragEnd(index)}
        />
      ))}
      <Polyline
        geometry={{
          coordinates: placemarks.map(placemark => get(placemark, 'geometry.coordinates'))
        }}
        options={{
          strokeColor: '#0000ff',
          strokeWidth: 4,
          strokeOpacity: 0.6
        }}
      />
    </Map>
  </YMaps>
);

CustomMap.propTypes = {
  placemarks: PropTypes.array,
  handleAddPlacemark: PropTypes.func,
  handleSelectSearchResult: PropTypes.func,
  addPlacemarkRef: PropTypes.func,
  handlePlacemarkDragEnd: PropTypes.func
};

CustomMap.defaultProps = {
  placemarks: []
};

export default CustomMap;
