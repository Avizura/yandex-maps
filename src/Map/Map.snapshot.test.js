import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Map from './Map';

describe('Map', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Map />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
