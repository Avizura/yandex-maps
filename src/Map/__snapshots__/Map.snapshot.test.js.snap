// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly 1`] = `
<div
  style={
    Object {
      "height": 600,
      "width": "100%",
    }
  }
/>
`;
